web/docs/deployment/database.md

```diff
@@ -2,10 +2,66 @@
 title: Database
 ---
 
-TODO: we'll explain the database in the context of deployment
+In this section, we'll discuss what happens with the database when your app goes live. When you develop your app locally, you probably use the dev database that you start with the `wasp db start` command. However, when it's time to deploy your app, you'll need to set up a production database.
 
-TODO: the only requirement from Wasp's point of view - it has to be a PostgreSQL database accessible from the server via the `DATABASE_URL` environment variable
+### Production database requirements
 
-TODO: we'll talk about migrations: how are they created in development, how are they applied in production, how to debug failed migrations
+The server app that Wasp generates uses a PostgreSQL database. The only requirement from Wasp's point of view is that the database is accessible from the server via the `DATABASE_URL` environment variable.
 
-TODO: we'll talk about how to open the DB studio in development and production
\ No newline at end of file
+It can be a PostgreSQL database running on the same server as the server app, or it can be a managed PostgreSQL database service like [Fly Postgres](https://fly.io/docs/postgres/), [AWS RDS](https://aws.amazon.com/rds/), or some other service.
+
+## Migrations
+
+Every time you make a change in your [Prisma schema](../data-model/prisma-file.md) e.g. adding a new model, changing a field type, etc., you need to create a migration. Migrations are some code that describes the change you made in the schema, and they are used to apply the change to the database.
+
+The benefit of migrations is that you can apply the same change to multiple databases, e.g. your local development database and your production database. 
+
+### Creating migrations
+
+After you made a change in the Prisma schema, you can create a migration by running the following command:
+
+```bash
+wasp db migrate-dev
+```
+
+This command will create a new migration in the `migrations` directory. The migration is a set of SQL commands that describe the change you made in the schema.
+
+### Applying migrations
+
+**In development**, the migrations are applied as soon as you run the `wasp start` command. 
+
+**In production**, the server app first checks if there are any new migrations that need to be applied, and if there are, it applies them before starting the server. This way, the database schema is always in sync with the Prisma schema.
```

Comment:
Actually it is not a server app doing this, it is our Dockerfile, right? I would make it clera this behaviour is present only if they use our Dockerfile.

Reply:
Well actually, it's the `start-production` npm script. It runs `npm run db-migrate-prod` before starting the server. I'll mention this.